src/ucp/wireup/select.c

```diff
@@ -74,27 +74,32 @@ typedef struct {
 
 
 /**
- * Context for lanes selection during UCP wireup procedure
+ * Global parameters for lanes selection during UCP wireup procedure
  */
 typedef struct {
     ucp_ep_h                  ep;                        /* UCP Endpoint */
-    const ucp_ep_params_t     *params;                   /* Tuning parameters for the
+    const ucp_ep_params_t     *ep_params;                /* Tuning parameters for the
                                                           * UCP endpoint */
     unsigned                  ep_init_flags;             /* Endpoint init flags */
-    uint64_t                  tl_bitmap;                 /* Global TL bitmap mask */
+    uint64_t                  tl_bitmap;                 /* TLs bitmap which can be selected */
     const ucp_address_entry_t *address_list;             /* Array of remote addresses */
     unsigned                  address_count;             /* Number of remote addresses */
-    ucp_wireup_lane_desc_t    lane_descs[UCP_MAX_LANES]; /* Array of active lanes that are
-                                                          * found during selection */
-    ucp_lane_index_t          num_lanes;                 /* Number of active lanes */
     int                       allow_am;                  /* Shows whether emulation over AM
                                                           * is allowed or not for RMA/AMO */
     int                       show_error;                /* Global flag that controls showing
                                                           * errors from a selecting transport
                                                           * procedure */
-    ucp_wireup_select_info_t  am_info;                   /* AM transport selection info */
-} ucp_wireup_select_ctx_t;
+} ucp_wireup_select_params_t;
```

Comment:
maybe keep am_info here?

Reply:
This structure is initialized once, am_info is a result of AM selection logic which goes further only to TM selection logic.